src/System.Threading/src/System/Threading/ReaderWriterLock.cs

```diff
@@ -0,0 +1,1339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.ConstrainedExecution;
+
+namespace System.Threading
+{
+    public sealed class ReaderWriterLock : CriticalFinalizerObject
+    {
+        private const int InvalidThreadID = -1;
+        private const ushort MaxAcquireCount = ushort.MaxValue;
+        private static readonly int DefaultSpinCount = Environment.ProcessorCount != 1 ? 500 : 0;
+
+        private static long s_mostRecentLockID;
+
+        private ManualResetEvent _readerEvent;
+        private AutoResetEvent _writerEvent;
+        private long _lockID;
+        private volatile int _state;
+        private int _writerID = InvalidThreadID;
+        private int _writerSeqNum;
+        private ushort _writerLevel;
+
+#if RWLOCK_STATISTICS
```

Comment:
If we think we would want to turn it into actual logging, e.g. via an EventSource, we could leave it, though I'm not sure we want to be investing in this particular class further.  Otherwise, yeah, I think it makes sense to clean it up.  Up to you, though.